libs/designer/src/lib/ui/templates/basics/resource.tsx

```diff
@@ -59,10 +60,9 @@ export const ResourceSection = (props: ResourceSectionProps) => {
 const WorkflowName = ({ workflowId, label, placeholder }: { workflowId: string; label: string; placeholder?: string }) => {
   const dispatch = useDispatch<AppDispatch>();
   const styles = useStyles();
-  const {
-    workflowName,
-    errors: { workflow: workflowError },
-  } = useWorkflowTemplate(workflowId);
+  const workflowTemplate = useWorkflowTemplate(workflowId);
+  const workflowName = useMemo(() => workflowTemplate?.workflowName, [workflowTemplate]);
+  const workflowError = useMemo(() => workflowTemplate?.errors?.workflow, [workflowTemplate]);
```

Comment:
I am assuming it is only this component because all other components render only when template data is present. Else it would be a bug and we will want to catch that.

Reply:
Yes this was the component running into the error over in AI Foundry